src/test/java/org/bonitasoft/connectors/scripting/ShellConnectorTest.java

```diff
@@ -11,182 +11,192 @@
-            ShellConnector shell = new ShellConnector();
-            final HashMap<String, Object> parametersMap = new HashMap<String, Object>(1);
-            shell.setInputParameters(parametersMap);
-            shell.validateInputParameters();
-        }
+    // =================================================================================================================
+    // UTILS
+    // =================================================================================================================
+
+    private static String defaultOsInterpreter() {
+        return isUnix() || isMac() ? "/bin/sh" : "cmd.exe";
     }
 
-    @Test
-    public void testRottenScript() throws Exception {
-        if (isMac() || isUnix() || isWindows()) {
-            ShellConnector shell = new ShellConnector();
-            final HashMap<String, Object> parametersMap = new HashMap<String, Object>(1);
-            parametersMap.put("interpreter", getInterpreter());
-            parametersMap.put("parameter", getParameter());
-            parametersMap.put("script", "hefv ghfvg ");
-            shell.setInputParameters(parametersMap);
-            shell.validateInputParameters();
-            shell.execute();
-            System.out.println("result:" + shell.getResult());
-            System.out.println("status:" + shell.getExitStatus());
-            assertThat((Integer) shell.getExitStatus(), not(is(0)));
+    private static String defaultOsParameter() {
+        return isUnix() || isMac() ? "-c" : "/c";
+    }
 
-        }
+    private static String listContentOfCurrentDirectoryShellScript() {
+        return isUnix() || isMac() ? "ls -lia" : "dir";
     }
 
-    @Test
-    public void testPipe() throws ConnectorValidationException, ConnectorException {
-        if (isMac() || isUnix()) {
-            ShellConnector shell = new ShellConnector();
-            final HashMap<String, Object> parametersMap = new HashMap<String, Object>(1);
-            parametersMap.put("interpreter", "/bin/sh");
-            parametersMap.put("parameter", "-c");
-            parametersMap.put("script", "ls | wc -l");
-            shell.setInputParameters(parametersMap);
-            shell.validateInputParameters();
-            shell.execute();
-            assertThat(shell.getExitStatus(), is(shell.getExitStatus()));
+    private static String multiLineShellScript() {
+        if (isUnix() || isMac()) {
+            return script("ls"
+                    , "var=2"
+                    , "echo $var"
+                    , "var=$(($var+1))"
+                    , "echo var"
+                    , "sleep 2");
+        } else {
+            return script("dir"
+                    , "set var=2"
+                    , "echo %var%"
+                    , "set /a var=%var%+1"
+                    , "echo %var%"
+                    , "ping 123.45.67.89 -n 1 -w 2500 > nul");
         }
     }
+
+    private static String script(String... elements) {
```

Comment:
could be renamed `multiline` or something more meaningful.

Reply:
ok, I will manage this in a refactoring PR